src/components/EditorsColumn.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import bindAll from 'lodash/bindAll';
+import isEmpty from 'lodash/isEmpty';
+import includes from 'lodash/includes';
+import partial from 'lodash/partial';
+
+import {
+  editorFocusedRequestedLine,
+  updateProjectSource,
+} from '../actions';
+
+import EditorContainer from './EditorContainer';
+import Editor from './Editor';
+
+class EditorsColumn extends React.Component {
+  constructor() {
+    super();
+    bindAll(
+      this,
+      '_handleEditorInput',
+      '_handleRequestedLineFocused',
+    );
+  }
+
+  _allErrorsFor(language) {
+    if (language === 'javascript') {
+      return this.props.errors.javascript.items.
+      concat(this.props.runtimeErrors);
+    }
+
+    return this.props.errors[language].items;
+  }
+
+  _handleEditorInput(language, source) {
+    this.props.actions.updateProjectSource(
+      this.props.currentProject.projectKey,
+      language,
+      source,
+    );
+  }
+
+  _handleRequestedLineFocused() {
+    this.props.actions.editorFocusedRequestedLine();
+  }
+
+  render() {
+    const {currentProject, ui} = this.props;
+    const editors = [];
+    ['html', 'css', 'javascript'].forEach((language) => {
+      if (includes(ui.minimizedComponents, `editor.${language}`)) {
+        return;
+      }
+
+      editors.push(
+        <EditorContainer
+          key={language}
+          language={language}
+          source={currentProject.sources[language]}
+          onMinimize={
+            partial(this.props.onComponentMinimize, `editor.${language}`)
+          }
+        >
+          <Editor
+            errors={this._allErrorsFor(language)}
+            key={language}
+            language={language}
+            percentageOfHeight={1 / editors.length}
+            projectKey={currentProject.projectKey}
+            requestedFocusedLine={ui.editors.requestedFocusedLine}
+            source={currentProject.sources[language]}
+            onInput={partial(this._handleEditorInput, language)}
+            onRequestedLineFocused={this._handleRequestedLineFocused}
+          />
+        </EditorContainer>,
+      );
+    });
+
+    if (isEmpty(editors)) {
+      return null;
+    }
+
+    return (
+      <div className="environment__column">
+        <div className="environment__columnContents editors">{editors}</div>
+      </div>
+    );
+  }
+}
+
+EditorsColumn.propTypes = {
+  actions: React.PropTypes.object.isRequired,
+  currentProject: React.PropTypes.object.isRequired,
+  errors: React.PropTypes.object.isRequired,
+  runtimeErrors: React.PropTypes.array.isRequired,
+  ui: React.PropTypes.object.isRequired,
+  onComponentMinimize: React.PropTypes.func.isRequired,
+};
+
+function mapDispatchToProps(dispatch) {
```

Comment:
@outoftime I hear you. I went this route for the sake of moving as much stuff out of `Workspace.jsx` as possible, but I can move the dispatches back and use callbacks. 

I could also put a `mapDispatchToProps` in `Workspace.jsx` while I'm at it...

As far as alternative structure... we could follow the "Containers" convention and move `Workspace` and `EditorsColumn` into a `containers/` directory reserved for components that are `connect`ed. This is a pretty minimal change (just moving two files and changing some imports) but I think people find it makes it easier to disambiguate top-level from display components.